File: 天气/tq.js
const api = "https://cn.apihz.cn/api/tianqi/tqybip.php?id=10018490&key=cdc294ee577ac2bd96bfa8651dab80b3";

function showWeather(data){
    var now = data.nowinfo || {};
    document.getElementById('weather').textContent = (data.weather1 === data.weather2)?data.weather1:(data.weather1+"转"+data.weather2);
    document.getElementById('temperature').textContent = now.temperature + "℃";
    document.getElementById('place').textContent = data.guo + " " + data.sheng + " " + data.shi;
    document.getElementById('uptime').textContent = now.uptime;
}

function fetcheather(){
    document.getElementById('status').textContent = "加载中...";
    fetch(api)
        .then(response => response.json())
        .then(result => {
            if(result.code === 200){
                showWeather(result);
                document.getElementById('status').textContent = "";
            }
            else{
                document.getElementById('status').textContent = "加载失败："+result.msg;
            }
        })
        .catch(error => {
            document.getElementById('status').textContent = "❌ 网络请求失败（" + error.message + "）";
        });
}

document.addEventListener('DOMContentLoaded', function() {
    fetcheather();
    setInterval(fetcheather, 60000);
    document.getElementById('refreshbtn').addEventListener('click', fetcheather);
});

File: 天气/tq.py
import requests
import time
from flask import Flask, jsonify
from flask_cors import CORS
app = Flask(__name__)
CORS(app)

api_url="https://cn.apihz.cn/api/tianqi/tqybip.php"
params = {
    "id": "10018490",
    "key": "cdc294ee577ac2bd96bfa8651dab80b3",
}

@app.route('/weather')
def get_weather():
    try:
        response = requests.get(api_url, params=params)
        data=response.json()
        if data.get("code") == 200:
            now=data.get("nowinfo",{})
            resultg={
                "place":f"{data.get('guo')} {data.get('sheng')} {data.get('shi')}",
                "temperature":now.get('temperature'),
                "weather1":data.get('weather1'),
                "weather2":data.get('weather2'),
                "uptime":now.get('uptime'),
            }
            return jsonify({"code":200,"data":resultg})
        else:
            return jsonify({"code":500,"msg":data.get("msg" ,"获取天气失败")})     
    except Exception as e:
        return jsonify({"code":500,"msg":str(e)})
    
if __name__ == '__main__':
    print("🌤️ 天气服务启动: http://localhost:5000/weather")
    print("浏览器打开 tq.html 即可查看")
    app.run(host='0.0.0.0', port=5000)